### PR TITLE
chore(argo-cd): Add `app.kubernetes.io/version` label

### PR DESCRIPTION
**Pull request by yu-croco**
resolves https://github.com/argoproj/argo-helm/issues/2118 🙋 

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-cd/templates/_common.tpl [hunk 1]
```diff
@@ -46,6 +46,7 @@ helm.sh/chart: {{ include "argo-cd.chart" .context }}
 {{ include "argo-cd.selectorLabels" (dict "context" .context "component" .component "name" .name) }}
 app.kubernetes.io/managed-by: {{ .context.Release.Service }}
 app.kubernetes.io/part-of: argocd
+app.kubernetes.io/version: {{ .Chart.AppVersion }}
```
Comment (yu-croco) [inline charts/argo-cd/templates/_common.tpl:49]: https://helm.sh/docs/chart_best_practices/labels/#standard-labels

image

Comment (yu-croco) [reply charts/argo-cd/templates/_common.tpl:49]: `argo-cd.labels` is used almost all of all resources, so I think we set it here. 🤔

File: charts/argo-cd/templates/_common.tpl [hunk 2]
```diff
@@ -46,6 +46,7 @@ helm.sh/chart: {{ include "argo-cd.chart" .context }}
 {{ include "argo-cd.selectorLabels" (dict "context" .context "component" .component "name" .name) }}
 app.kubernetes.io/managed-by: {{ .context.Release.Service }}
 app.kubernetes.io/part-of: argocd
+app.kubernetes.io/version: {{ include "argo-cd.defaultTag" . }}
 {{- with .context.Values.global.additionalLabels }}
```
Comment (yu-croco) [inline charts/argo-cd/templates/_common.tpl:50]: https://helm.sh/docs/chart_best_practices/labels/#standard-labels
image

`"argo-cd.labels"` is used to all of resources, so I think it's ok to add this label here. 🙋

**Review by tico24 — APPROVED** [review]

**Review by psimms-r7 — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]

**Review by jmeridth — APPROVED** [review]

**Comment by sathieu** [comment]
This [fails](https://gitlab.com/kubitus-project/kubitus-installer/-/jobs/4603604250) when the tag includes the digest :

`metadata.labels: Invalid value: "v2.7.6@sha256:7daba5f38b23f4f091951b727db6f87dc04ad396fd21044401502438d633836e": must be no more than 63 characters, metadata.labels: Invalid value: "v2.7.6@sha256:7daba5f38b23f4f091951b727db6f87dc04ad396fd21044401502438d633836e": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')`

I'll propose a PR to fix this.

**Comment by sathieu** [comment]
Actually, this was already discussed https://github.com/argoproj/argo-helm/pull/1427#issuecomment-1228825486.

And the conclusion was to drop the label. What should I do now?

**Comment by sathieu** [comment]
Here is my proposed solution: https://github.com/argoproj/argo-helm/pull/2148

**Pull request merged**